sh_scrapy/settings.py

```diff
@@ -75,7 +95,10 @@ def _populate_settings_base(apisettings, defaults_func, spider=None):
     job_settings = apisettings.setdefault('job_settings', {})
 
     defaults_func(s)
-    _load_addons(enabled_addons, s, o)
+    on_missing_addons = _get_action_on_missing_addons([
+        job_settings, spider_settings,
+        organization_settings, project_settings])
```

Comment:
why addons are setup before settings are merged? maybe it's possible to merge settings first and then set all defaults with lower priority. I think you can use `scrapy.settings.BaseSettings` here for that purpose instead of plain `dict` - then you'll be able to use different priorities for different settings (e.g. `10` for project settings, `20` for org, `30` for spider, `40` for job settings) and then check value of 'ON_MISSING_ADDONS' and update settings object with base addon settings with priority `0`. then merge result `o` with original settings `s` with priority 'cmdline' - the same way it's done now.